backend/loging/views.py

```diff
@@ -0,0 +1,171 @@
+                central_existing_json_data.append(data)
+
+                central_updated_content = json.dumps(
+                    central_existing_json_data, cls=CustomJSONEncoder
+                )
+                central_blob_client.upload_blob(central_updated_content, overwrite=True)
+
+                return Response(
+                    {"message": "Data stored in Azure Blob successfully"},
+                    status=status.HTTP_201_CREATED,
+                )
+            except Exception as e:
+                return Response(
+                    {"message": "Failed to store data in Azure Blob", "error": str(e)},
+                    status=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                )
+        else:
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+    def get(self, request):
+        try:
+            user_id = request.query_params.get("user_id")
+            start_date_str = request.query_params.get("start_date")
+            end_date_str = request.query_params.get("end_date")
+
+            if user_id is None:
+                return Response(
+                    {"message": "user_id is required"},
+                    status=status.HTTP_400_BAD_REQUEST,
+                )
+
+            try:
+                user = User.objects.get(id=user_id)
+            except User.DoesNotExist:
+                return Response(
+                    {"message": "User with the provided user_id does not exist"},
+                    status=status.HTTP_404_NOT_FOUND,
+                )
+
+            if start_date_str and end_date_str:
+                task = retrieve_logs.delay(start_date_str, end_date_str)
+            else:
+                task = retrieve_logs.delay(None, None)
+
+            result = task.get()
+            log_content = result.get("log_content", "")
+            file_name = result.get("file_name", "")
+            file_name_with_prefix = f"temp_{file_name}"
+
+            blob_service_client = BlobServiceClient.from_connection_string(
+                AZURE_STORAGE_CONNECTION_STRING
+            )
+            container_client = blob_service_client.get_container_client(CONTAINER_NAME)
+            blob_client = container_client.get_blob_client(file_name_with_prefix)
+            blob_client.upload_blob(log_content, overwrite=True)
+
+            expiry = datetime.datetime.utcnow() + datetime.timedelta(hours=1)
+            sas_token = generate_blob_sas(
+                container_name=CONTAINER_NAME,
+                blob_name=blob_client.blob_name,
+                account_name="shoonyastoragedevelop",
+                account_key=extract_account_key(AZURE_STORAGE_CONNECTION_STRING),
+                permission=BlobSasPermissions(read=True),
+                expiry=expiry,
+            )
+
+            blob_url = f"https://shoonyastoragedevelop.blob.core.windows.net/{CONTAINER_NAME}/{blob_client.blob_name}?{sas_token}"
```

Comment:
We should store the URL as env variable ? 
@KunalTiwary @suyogkokaje

Reply:
This is a generic URL, the main/hidden components are taken from env. However, if you suggest we can keep this as well.